src/bp/core/services/telemetry/hooks.ts

```diff
@@ -0,0 +1,102 @@
+import LicensingService from 'common/licensing-service'
+import { buildSchema } from 'common/telemetry'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { GhostService } from '..'
+import { BotService } from '../bot-service'
+import { JobService } from '../job-service'
+
+import { TelemetryStats } from './telemetry-stats'
+
+interface BotHooks {
+  botId: string
+  after_bot_mount?: number
+  after_bot_unmount?: number
+  after_event_processed?: number
+  after_incoming_middleware?: number
+  before_bot_import?: number
+  before_incoming_middleware?: number
+  before_outgoing_middleware?: number
+  before_session_timeout?: number
+  before_suggestions_election?: number
+  on_bot_error?: number
+}
+
+interface GlobalHooks {
+  after_server_start?: number
+  after_stage_changed?: number
+  on_incident_status_changed?: number
+  on_stage_request?: number
+}
+interface HookPayload {
+  perBots: BotHooks[]
+  global: GlobalHooks
+}
```

Comment:
Specifying the name is not needed. It adds another list that needs to be updated. Make it dynamic.

```js
interface HookPayload {
  perBots: { botId: string } & {
    [hookName: string]: number
  }
  global: { [hookName: string]: number }
}
```

Reply:
fixed in 8282c80ceb579ed1492e5793651ab9633ce9ba44